assets/chat/js/chat.js

```diff
@@ -540,7 +540,7 @@ class Chat {
 
         if(win === null)
             win = this.mainwindow
-        win.lock()
+        if(!this.backlogloading) win.lock();
```

Comment:
Curious: if we end up _not_ getting this lock, do we want to ensure that the code between here and the unlock is _not_ executed? As in, should we use `win` at all without this lock?

Reply:
The ChatWindow adds the messages to the actual DOM, so in that sense we need to use it.
And seeing as `updateAndPin` is called after inserting the history, I don't think the locking matters.